src/nsidc/iceflow/itrf/converter.py

```diff
@@ -69,17 +71,50 @@ def transform_itrf(
             if not plate:
                 plate = plate_name(Point(chunk.longitude.mean(), chunk.latitude.mean()))
             plate_model_step = (
-                f"+step +inv +init={target_itrf}:{plate} +t_epoch={target_epoch} "
+                # Perform coordinate propagation to the given epoch using the
+                # provided plate motion model (PMM).  An example is given in the
+                # message of this commit:
+                # https://github.com/OSGeo/PROJ/commit/403f930355926aced5caba5bfbcc230ad152cf86
+                f"+step +init={target_itrf}:{plate} +t_epoch={target_epoch} "
             )
 
+        itrf_transformation_step = ""
+        if source_itrf != target_itrf:
+            # This performs a helmert transform (see
+            # https://proj.org/en/9.4/operations/transformations/helmert.html). `+init=ITRF2014:ITRF2008`
+            # looks up the ITRF2008 helmert transformation step in the ITRF2014
+            # data file (see e.g.,
+            # https://github.com/OSGeo/PROJ/blob/master/data/ITRF2014). The
+            # `+inv` reverses the transformation. So `+init=ITRF2014:ITRF2008`
+            # performs a helmert transform from ITRF2008 to ITRF2014.
+            itrf_transformation_step = f"+step +inv +init={target_itrf}:{source_itrf} "
+
         pipeline = (
+            # This initializes the pipeline and declares the use of the WGS84
+            # ellipsoid for all of the following steps. See
+            # https://proj.org/en/9.5/operations/pipeline.html.
             f"+proj=pipeline +ellps=WGS84 "
+            # Performs unit conversion from lon/lat degrees to radians.
+            # TODO: This step appears to be unnecessary. Removing it does not appear to
+            # affect the output. The following steps require that the
+            # coordinates be geodedic, which could be radians or degrees.
             f"+step +proj=unitconvert +xy_in=deg +xy_out=rad "
```

Comment:
I did some testing and this step does not appear to be necessary. Proj seems to assume lat/lon inputs by default and is "unit-aware" in that it will raise an error between steps if it detects an incompatibility. 

I don't think it hurts to be here, but if not necessary I'm inclined to remove it to simplify the transformation.

```
$ echo "1 1 1 1" | cct -v -d 14 +proj=pipeline +ellps=WGS84 +step +proj=unitconvert +xy_in=deg +xy_out=rad +step +proj=latlon +step +proj=cart +step +inv +init=ITRF2014:ITRF2008 +step +init=ITRF2014:ANTA +t_epoch=2019.7 +step +inv +proj=cart +step +proj=unitconvert +xy_in=rad +xy_out=deg
0.99961901119316  0.99981772606697  1.37719855178148        1.0000
```
is the same as:
```
$ echo "1 1 1 1" | cct -v -d 14 +proj=pipeline +ellps=WGS84 +step +proj=latlon +step +proj=cart +step +inv +init=ITRF2014:ITRF2008 +step +init=ITRF2014:ANTA +t_epoch=2019.7 +step +inv +proj=cart
0.99961901119316  0.99981772606697  1.37719855178148        1.0000
```

Reply:
I think we got the outline from our pipeline from this response from Even Rouault  in a proj thread that I started: https://lists.osgeo.org/pipermail/proj/2019-May/008509.html

In [a subsequent message](https://lists.osgeo.org/pipermail/proj/2019-May/008512.html) Alan Snow notes:

>  AFAIU, units are now degrees as in EPSG:4326, and in this case a conversion to radians is no longer required, as in proj-4.

Which would seem to suggest that the unit conversion step could be skipped.

Reply:
The examples given in this PR to add the ITRF2020 params to proj shows the unit conversion steps. This may be a best-practice. See https://github.com/OSGeo/PROJ/pull/4235.